public/app/features/dimensions/editors/ResourcePickerPopover.tsx

```diff
@@ -0,0 +1,273 @@
+            {mediaType === 'image' && newValue && <img src={imgSrc} className={styles.img} />}
+          </div>
+        </Field>
+        <Label>{shortName}</Label>
+      </div>
+    </>
+  );
+
+  const renderPicker = () => {
+    switch (activePicker) {
+      case 'folder':
+        return renderFolderPicker();
+      case 'url':
+        return renderURLPicker();
+      default:
+        return renderFolderPicker();
+    }
+  };
+
+  return (
+    <FocusScope>
+      <div className={styles.resourcePickerPopover}>
+        <div className={styles.resourcePickerPopoverTabs}>
+          <button className={getTabClassName('folder')} onClick={() => setActivePicker('folder')}>
+            Folder
+          </button>
+          <button className={getTabClassName('url')} onClick={() => setActivePicker('url')}>
+            URL
+          </button>
+        </div>
+        <div className={styles.resourcePickerPopoverContent}>
+          {renderPicker()}
+          <Button
+            className={styles.selectButton}
+            variant={newValue && newValue !== value ? 'primary' : 'secondary'}
+            onClick={() => onChange(newValue)}
+          >
+            Select
+          </Button>
+        </div>
+      </div>
+    </FocusScope>
+  );
+};
+
+const getStyles = stylesFactory((theme: GrafanaTheme2) => {
+  return {
+    cardsWrapper: css`
+      height: 30vh;
+      min-height: 50px;
+      margin-top: 5px;
+      max-width: 680px;
+    `,
+    iconPreview: css`
+      width: 238px;
+      height: 198px;
+      border: 1px solid ${theme.colors.border.medium};
+      display: flex;
+      align-items: center;
+      justify-content: center;
+    `,
+    iconContainer: css`
+      display: flex;
+      flex-direction: column;
+      width: 80%;
+      align-items: center;
+      align-self: center;
+    `,
+    img: css`
+      width: 147px;
+      height: 147px;
+      fill: ${theme.colors.text.primary};
+    `,
+    resourcePickerPopover: css`
+      border-radius: ${theme.shape.borderRadius()};
+      box-shadow: ${theme.shadows.z3};
+      background: ${theme.colors.background.primary};
+      border: 1px solid ${theme.colors.border.medium};
+
+      .ResourcePickerPopover__tab {
+        width: 50%;
+        text-align: center;
+        padding: ${theme.spacing(1, 0)};
+        background: ${theme.colors.background.secondary};
+        color: ${theme.colors.text.secondary};
+        font-size: ${theme.typography.bodySmall.fontSize};
+        cursor: pointer;
+        border: none;
+
+        &:focus:not(:focus-visible) {
+          outline: none;
+          box-shadow: none;
+        }
+
+        :focus-visible {
+          position: relative;
+        }
+      }
+
+      .ResourcePickerPopover__tab--active {
+        color: ${theme.colors.text.primary};
+        font-weight: ${theme.typography.fontWeightMedium};
+        background: ${theme.colors.background.primary};
+      }
```

Comment:
You don't need nesting in emotion, I would suggest going with a flat classes definition unless nesting serves here any purpose.